src/main/java/com/academy/edge/studentmanager/controllers/GradeController.java

```diff
@@ -13,15 +17,26 @@
 
 import java.util.List;
 
+
 @RestController
 @RequestMapping("/api/v1/grades")
 public class GradeController {
+
     final GradeService gradeService;
 
+    @Autowired
     public GradeController(GradeService gradeService) {
         this.gradeService = gradeService;
     }
 
+    @PutMapping
+    @PreAuthorize("hasAnyRole('ADMIN','INSTRUCTOR', 'STUDENT')")
```

Comment:
O estudante deveria apenas conseguir alterar a própria nota, pode utilizar os outros métodos como exemplo
```suggestion
    @PreAuthorize("hasAnyRole('ADMIN','INSTRUCTOR', 'STUDENT')")
```